### ISSUE TITLE
Using `eachindex/axes` suggestion should not be given for ranges that don't start from 1

### ISSUE THREAD

**Issue by Datseris** [julia-vscode/julia-vscode#2973]
Example:

![image](https://user-images.githubusercontent.com/19669089/181235545-5296cb08-5df6-4c74-aafa-1b150c1255a7.png)

While I agree on suggesting `eachindex` to replace `1:length(A)`, I don't think it is simpler, or more intuitive, to use `axes/eachindex` to replace `2:length(A)`. Actually, it isn't even possible, right?

**Comment by pfitzseb** [comment]
Sure that's possible:
```
julia> data = [1 2 3;
               4 5 6]
2×3 Matrix{Int64}:
 1  2  3
 4  5  6

julia> [data[:, i] for i in 2:size(data, 2)]
2-element Vector{Vector{Int64}}:
 [2, 5]
 [3, 6]

julia> [data[:, i] for i in Iterators.drop(axes(data, 2), 1)]
2-element Vector{Vector{Int64}}:
 [2, 5]
 [3, 6]
```

**Comment by Datseris** [comment]
Yes, you are right, you can in this way, but that's not what I was thinking :D 

I could be just dead wrong and `Iterators.drop(axes(data, 2), 1)` is more intuitive to `2:size(data, 2)` to the average user. That's provided of course that the average user already knows of `Iterators.drop` as intimately as `size`, so that they don't have to google "how to skip elements Julia" once they see the VSCode suggestion.

**Comment by Datseris** [comment]
(note: I am fully aware of the new movement of stopping to use 1-based indexing when writing source code which is of course a good idea. What I am putting up for debate is whether this recommendation should be given to the general VSCOde user and in situations like this, that already show a kinda clear purposeful utilizatioon of the 1-based indexing)

**Comment by pfitzseb** [comment]
I'm not trying to say it's more intuitive; it's just more *correct* in the general case :)

That said, this is a false positive in a sense; AFAICT `readdlm` always returns a `Matrix`, for which one-based indexing is correct. Of course, neither Julia nor the linter know about that. An easy fix for this is to only apply this annotation if we know the type of array (and it's neither `Any` nor a `Base.Array`).

**Comment by davidanthoff** [comment]
Have we made this lint an opt-in? I kind of think anything that can produce false positives should be opt-in.

**Comment by gizzye** [comment]
Is it possible to turn off this warning ?

**Comment by Datseris** [comment]
Just to be clear: the trend still exists in Julia to "not explicitly use 1 based indexing", however my later statement that "is of course a good idea" I have to completely retract. I have now almost entirely gone against this movement to make my code nice and simple and I've stopped caring about the 0.001% of the community that uses offset arrays or whatever thing in Julia does not have 1-based indexing (nothing I've ever seen in any code).

**Comment by arismavridis** [comment]
Thanks for the reply, George. 
To make my question clear: do we know what would be the reason to avoid 1 based indexing? 
Would that benefit the compiler in some way, or is it just to make the syntax more "agreeable" for people used to 0-based indexing? 
Genuinely curious about that.

**Comment by pfitzseb** [comment]
Not every AbstractArray is using one-based indexing, so generic code (i.e. library functions that don't specialize on `Array`) should  instead use the generic interface suggested here. Of course that's not particularly relevant for private code, for which you know all possible use cases.

**Comment by arismavridis** [comment]
That's clear, thanks a lot for the reply!

**Comment by davidanthoff** [comment]
The language server backend was rewritten in v1.220.0: StaticLint.jl and SymbolServer.jl were folded into JuliaWorkspaces, and completion, hover, signature help, go-to-definition and find-references were reimplemented. This report predates that rewrite, so the code that produced it no longer exists and the behaviour cannot be confirmed against the current engine.

Closing as outdated. If you still see this on a current release, please open a fresh issue with a reproducer and we will look at it against the new engine.
